### PR TITLE
fix: table border issue

### LINKED ISSUE #38724: table 合并列样式错误（最后一行缺少border）
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/biao-ge-xing-lie-he-bing-antd-5-0-0-forked-roxl90)

### Steps to reproduce

合并列必现

### What is expected?

border-bottom可以正常设置

### What is actually happening?

缺少了border

| Environment | Info |
| --- | --- |
| antd | 5.0.0 |
| React | 18 |
| System | mac |
| Browser | chorme107 |

### LINKED ISSUE #39728: 表格行列合并后，合并行的边框缺失
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/biao-ge-xing-lie-he-bing-antd-5-1-0-forked-nrcm2l)

### Steps to reproduce

无

### What is expected?

合并后存在边框

### What is actually happening?

合并后边框不存在

| Environment | Info |
| --- | --- |
| antd | 5.1.0 |
| React | react@18.2.0 |
| System | macOS 13.1 |
| Browser | 108.0.5359.124 |

image

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/38724
close https://github.com/ant-design/ant-design/pull/39677
close https://github.com/ant-design/ant-design/issues/39728
close https://github.com/ant-design/ant-design/pull/38735

### 💡 Background and solution
顺便把 border 改成实色，解决一下重叠之后颜色加深的问题

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Table missing `border-bottom` in merged cell and unexpected border color if overlapping。         |
| 🇨🇳 Chinese |    修复 Table 组件合并单元格后底部边框消失和边框重叠时颜色变深的问题。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
argos 有点奇怪：https://ik.imagekit.io/argos/bfa8e6eae75493577da7372315b867b4486f7acb169efb4d4cae15604595274c

**Comment by MadCcc** [comment]
> argos 有点奇怪：https://ik.imagekit.io/argos/bfa8e6eae75493577da7372315b867b4486f7acb169efb4d4cae15604595274c

之前的 border 时透明色，暗色主题下就是白色带透明度，所以白底上看不出来。实际应该是 table 的 border radius 问题。

**Comment by MadCcc** [comment]
> argos 有点奇怪：https://ik.imagekit.io/argos/bfa8e6eae75493577da7372315b867b4486f7acb169efb4d4cae15604595274c

这个 demo 我改了

**Comment by MadCcc** [comment]
argos 响应有点慢

**Comment by afc163** [comment]
> 顺便把 border 改成实色，解决一下重叠之后颜色加深的问题

changelog 也写一下。

**Comment by MadCcc** [comment]
> > 顺便把 border 改成实色，解决一下重叠之后颜色加深的问题
> 
> changelog 也写一下。

Done.

**Review by afc163 — APPROVED** [review]

File: components/table/style/index.tsx [hunk 1]
```diff
@@ -170,11 +170,39 @@ const genTableStyle: GenerateStyle<TableToken, CSSObject> = (token) => {
       },
 
       // ============================ Body ============================
+      [`${componentCls}:not(${componentCls}-bordered)`]: {
```
Comment (zombieJ) [inline components/table/style/index.tsx:173]: 加点备注，现在 Table 样式特别复杂。得留条线索。

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
图片

这个会造成筛选表格时 border 的 transition 效果出现黑条。

**Pull request merged**